### DISCUSSION TITLE
Possible to add more motorsports?

### DISCUSSION THREAD

**Discussion by GRClark** [vasqued2/ha-teamtracker#56] [Ideas]
Would love to see more motorsports added such as NASCAR if at all possible.

**Comment by imist** [comment]
NASCAR should already be working although it doesn’t show anything right now since there is no race until feb 5th.
Just enter “NASCAR” as league and “*” as team/athlete when adding a new TeamTracker integration.

**Reply by Kappawaii** [reply]
Hey, i'm trying the make the API work and i'm getting "not found" doing exactly what u described above in this post

**Comment by vasqued2** [comment]
Just wanted to give a quick update.  The API is returning event info for the Clash at The Coliseum, however at this point, the field is not set and so it isn't finding any competitors.  

F1 generally treats a race as a multi-day event that includes the qualifying races.  It appears as though NASCAR is being treated as a single event on Sunday.  

It's possible that the competitors will still be populated and it will give live updates for qualifying, however my guess is that it will wait until after qualifying to load the competitors for the race and then update the race on Sunday.

Since I didn't find the NASCAR API until after the 2022 season was over, I'm not exactly sure how it will behave until Sunday.

**Comment by vasqued2** [comment]
NASCAR is working within the limitations of the API.  The most noticeable issues are:

- The competitors are not populated until after qualifying is completed (F1 provides real-time updates of qualifying too)
- The API does not return the Current Lap (F1 updates the current lap throughout the race)
- The API show the start time as the broadcast time.  The actual race can start up to 30 minutes later.  This can cause the initial real-time update to be delayed if the race starts more than 20 minutes after the posted start time.

Let me know if you notice any other problems.

**Comment by Kappawaii** [comment]
Hey guys, i'm trying to make this integration work on Nascar/F1/Indycar(IRL).
I'm having the same issue on F1 and NASCAR, both show "NOT FOUND" with * in the athlete field;
Indycar(IRL) shows POST since it's postseason, but both the other ones don't work :(

Any and all help would be much appreciated :)

**Reply by vasqued2** [reply]
See the post just above yours on the limitations of The NASCAR API.  It doesn’t return results until after qualifying is complete and the field is set.

this is a limitation of the API and I can’t do anything about it.  Other limitations are also noted.
